agent/xds/clusters.go

```diff
@@ -182,10 +182,20 @@ func (s *Server) clustersFromSnapshotMeshGateway(cfgSnap *proxycfg.ConfigSnapsho
 	return clusters, nil
 }
 
-func (s *Server) clustersFromServicesAndResolvers(
-	cfgSnap *proxycfg.ConfigSnapshot,
-	services map[structs.ServiceID]structs.CheckServiceNodes,
-	resolvers map[structs.ServiceID]*structs.ServiceResolverConfigEntry) ([]proto.Message, error) {
+func (s *Server) makeGatewayServiceClusters(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
```

Comment:
I don't know that I really like how these function signatures ended up. Open to suggestions.

Reply:
My main concern was passing down a nil service pointer to `makeGatewayCluster` when making clusters for destinations that aren't services, like datacenters.

I didn't know about the functional options pattern. Thanks for sharing!